### PR TITLE
2310/create schemas and models

### PR DESCRIPTION
**Pull request by aleneagu96**

### CODE REVIEW

**Review by andregeorgi — APPROVED** [review]

File: backend/.env.sample [hunk 1]
```diff
@@ -0,0 +1 @@
+// add the db name at the end of the URI
```
Comment (sorincretescu) [inline backend/.env.sample:1]: What is this here? A new file with a comment on it? =)) 
It should contain:
URI= //or whatever your variable is called

File: backend/logic/todoLogic.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+const mongoose = require("mongoose");
+const todoSchema = require("../../database/schemas");
+const Todo = mongoose.model("Todo", todoSchema);
+
+async function getDataFromDatabase() {
```
Comment (sorincretescu) [inline backend/logic/todoLogic.js:5]: Please start using arrow functions as a standard.

File: backend/logic/todoLogic.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    return data ?? [];
+  } catch (error) {
+    console.log("Error fetching data from the database", error);
+    throw error;
+  }
+}
+
+async function addDataToDatabase() {
+  try {
+    const newTodo = new Todo({
```
Comment (sorincretescu) [inline backend/logic/todoLogic.js:17]: What is this doing in a PR? We do not push "dummy code" unless it is necessary for a demo on the repo.

File: backend/logic/userLogic.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+async function getAllUsers() {
+    try {
+      await client.connect();
```
Comment (sorincretescu) [inline backend/logic/userLogic.js:3]: Why do you connect to the client in the "getAllUsers" function? We connect to the client, then we fire up requests. How about you are already logged in in the app ( suppose you implemented a persistent login ) and you wanna get all  the todos for a user and you do not make this request?

File: backend/logic/userLogic.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      const database = client.db("user");
```
Comment (sorincretescu) [inline backend/logic/userLogic.js:4]: Also, I thought we gave up on this kind of usage and we are starting on the standard, model-based way, as we've lastly talked.

File: backend/logic/userLogic.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    } catch (error) {
+      console.log("Error fetching data from the database", error);
+      throw error;
+    }
+  }
+
+  app.get("/", async (req, res) => {
+    try {
+      const data = await getAllUsers();
+      console.log(data);
```
Comment (sorincretescu) [inline backend/logic/userLogic.js:18]: No console logs unless they are for errors in the codebase.

File: backend/server.js [hunk 7]
```diff
@@ -1,52 +1,59 @@
...
+};
 
-async function getDataFromDatabase() {
+(async () => {
   try {
-    await client.connect();
-    const database = client.db("Todos");
-    const collection = database.collection("Todos");
+    await mongoose.connect(uri, mongooseOptions);
+    console.log("Connected to MongoDB");
```
Comment (sorincretescu) [inline backend/server.js:20]: You can leave this here for dev purposes.

File: backend/server.js [hunk 8]
```diff
@@ -1,52 +1,59 @@
...
+    try {
+      await addDataToDatabase();
+      res.json({ message: "Data added successfully" });
+    } catch (error) {
+      console.error("Error handling GET request", error);
+      res.status(500).json({ message: "Internal Server Error" });
+    }
+  });
+
+  const PORT = 5000;
```
Comment (sorincretescu) [inline backend/server.js:55]: Maybe also add the port to the .env file

File: database/models/index.js [hunk 9]
```diff
@@ -0,0 +1,3 @@
+const Todo = require('./todoModel');
+
+module.exports = Todo;
```
Comment (sorincretescu) [inline database/models/index.js:3]: This should be module.exports = {Todo}, because in the future you will also have other models there. This way, you will not have to change the code in multiple places at once, because you will already be treating it like it's an object with models.

File: database/models/userModel.js [hunk 10]
```diff
@@ -0,0 +1,6 @@
+import mongoose from "mongoose";
+
+ const User = mongoose.model("User", {userSchema});
+
+ const user = new User({username:'test', password:'test'});
```
Comment (sorincretescu) [inline database/models/userModel.js:5]: Also why is this here? Every time you start the project and fire up the db connection, it will try to save the same user.

File: database/schemas/todoSchema.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
+const mongoose = require("mongoose");
+const { Schema } = mongoose;
+
+
+const todoSchema = new Schema({
+    priority: Number,
+    description: String,
+    subtasks:[String],
```
Comment (sorincretescu) [inline database/schemas/todoSchema.js:8]: Spacing

File: database/schemas/todoSchema.js [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+}, {timestamps: true}
+);
+
+
+module.exports=todoSchema;
```
Comment (sorincretescu) [inline database/schemas/todoSchema.js:13]: Spacing

**Review by sorincretescu — CHANGES_REQUESTED** [review]

File: backend/logic/todoLogic.js (hunk 3 above)
Comment (aleneagu96) [reply backend/logic/todoLogic.js:17]: removed it

**Review by sorincretescu — APPROVED** [review]
